blocks/footer/footer.js

```diff
@@ -9,13 +9,67 @@ export default async function decorate(block) {
   const footerMeta = getMetadata('footer');
   block.textContent = '';
 
-  // load footer fragment
   const footerPath = footerMeta.footer || '/footer';
   const fragment = await loadFragment(footerPath);
 
-  // decorate footer DOM
   const footer = document.createElement('div');
   while (fragment.firstElementChild) footer.append(fragment.firstElementChild);
 
+  // Directly select the .footer-main container to append the columns to
+  const footerMain = footer.querySelector('.footer-main');
+  const section = footer.querySelector('.section');
+
+  // Find each <h3> within .footer-main
+  const mainSections = footerMain.querySelectorAll('h3');
+  mainSections.forEach((h3) => {
+    const nextUl = h3.nextElementSibling?.tagName === 'UL' ? h3.nextElementSibling : null;
+
+    const wrapperDiv = document.createElement('div');
+    wrapperDiv.classList.add('footer-column');
+
+    wrapperDiv.appendChild(h3);
+    if (nextUl) {
+      wrapperDiv.appendChild(nextUl);
+      h3.classList.add('toggle');
+      nextUl.classList.add('hidden'); // Initially hide the `ul` elements
```

Comment:
This is fine with me, but one thing to consider is to make things hidden by default in the CSS, and then visible only when a class is applied.  That way, they don't show up before your code executes.  This avoids CLS issues.

Reply:
I understand your point here. I refactored to make this hidden by default